kanidm_unix_int/src/daemon.rs

```diff
@@ -542,13 +543,10 @@ async fn main() {
             match task_listener.accept().await {
                 Ok((socket, _addr)) => {
                     // Did it come from root?
-                    match socket.peer_cred() {
-                        Ok(ucred) if ucred.uid() == 0 => (), // all good!
-                        _ => {
-                            // move along.
-                            debug!("Task handler not running as root, ignoring ...");
-                            continue;
-                        }
+                    if !matches!(socket.peer_cred(), Ok(ucred) if ucred.uid() == 0) {
+                        // is this too obfuscated?
```

Comment:
It does seem pretty cognitively complex...

Reply:
Yeah, I think this needs a clean up.